### PR TITLE
[WIP] Gradle: switch from 'jcenter()' to 'mavenCentral()' [JCenter will not accept new artifacts after March 31st 2021]

### PR DESCRIPTION
**Pull request by dejan2609**
Rationale: #10906

Few external links:
- https://jfrog.com/blog/into-the-sunset-bintray-jcenter-gocenter-and-chartcenter
- https://www.infoq.com/news/2021/02/jfrog-jcenter-bintray-closure
- https://blog.sonatype.com/dear-bintray-and-jcenter-users-heres-what-you-need-to-know-about-the-central-repository

### CODE REVIEW

**Comment by dejan2609** [comment]
Note: I will force push last commit only, just to add signing (of course, at the end all commits will be squashed into one).

**Comment by dejan2609** [comment]
Another commit pushed ('colorpicker' version upgrade, as suggested by @Gericop here: https://github.com/Gericop/ColorPicker/issues/6#issuecomment-774523682).

Note: commit was tested in isolation first (on top of the current master branch). 
In-detailed (overzealous ? 😅) git commit message contains all related details :bowtie:

**Comment by dejan2609** [comment]
Just to sum up current state (all details can be found on a related ticket): 
- it seems that patched android studio/android gradle plugin is on the way (I assume it will be a version 4.1.3): https://issuetracker.google.com/issues/179291081
- I will integrate few more libraries (exclusively published to a JCenter) directly to a project structure/gradle build (using `flatDir` gradle directive). Of course, it would much better  if those maintainers respond and agree to re-publish their libraries into Maven Central.... lets wait and see.

**Comment by dejan2609** [comment]
ℹ️ Herewith some more findings... it seems that issues with projects not being able to refresh (after JCenter is replaced with Maven Central) occurs only for projects that are using **Kotlin 1.3** while projects with **Kotlin 1.4** have no problems with refreshing whatsoever...

**Scenario (with much more details):**

1. clone project that uses **Kotlin 1.4** -->> **android/sunflower** https://github.com/android/sunflower

Checkout my PR: https://github.com/android/sunflower/pull/696 (that removes references to JCenter in favor of Maven Central) and execute: 
`./gradlew --refresh-dependencies --info`

Version for `jetbrains trove4j` is resolved to `1.0.20181211` (and this version can be found on a Maven Central) so everything is OK here ✅:

```
Resource missing. [HTTP HEAD: https://dl.google.com/dl/android/maven2/org/jetbrains/kotlin/kotlin-script-runtime/1.4.21/kotlin-script-runtime-1.4.21.pom]
Resource missing. [HTTP HEAD: https://dl.google.com/dl/android/maven2/org/jetbrains/intellij/deps/trove4j/1.0.20181211/trove4j-1.0.20181211.pom]
Resource missing. [HTTP HEAD: https://dl.google.com/dl/android/maven2/org/jetbrains/kotlin/kotlin-daemon-embeddable/1.4.21/kotlin-daemon-embeddable-1.4.21.pom]
Cached resource https://repo.maven.apache.org/maven2/org/jetbrains/intellij/deps/trove4j/1.0.20181211/trove4j-1.0.20181211.pom is up-to-date (lastModified: Tue Dec 11 13:54:51 CET 2018).
Cached resource https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-script-runtime/1.4.21/kotlin-script-runtime-1.4.21.pom is up-to-date (lastModified: Mon Dec 07 12:15:51 CET 2020).
Cached resource https://repo.maven.apache.org/maven2/org/jetbrains/kotlin/kotlin-daemon-embeddable/1.4.21/kotlin-daemon-embeddable-1.4.21.pom is up-to-date (lastModified: Mon Dec 07 12:14:42 CET 2020).
```
---

2. now, clone project that uses ***Kotlin 1.3*** -->> **android/architecture-components-samples** and checkout main branch: https://github.com/android/architecture-components-samples/tree/main 
Then go to 'BasicRxJavaSample' folder/module and refresh dependencies: 
`cd BasicRxJavaSample`
`./gradlew --refresh-dependencies --info` 

Version for `jetbrains trove4j` is resolved to `20160824` (please note this version was published to a JCenter ***only***). 
Build will pass ✅ (given a fact that this is a main branch that still has reference to a JCenter):

```
Cached resource https://jcenter.bintray.com/org/jetbrains/kotlin/kotlin-stdlib-jdk7/1.3.72/kotlin-stdlib-jdk7-1.3.72.jar is up-to-date (lastModified: Tue Apr 14 12:39:20 CEST 2020).
Cached resource https://jcenter.bintray.com/org/jetbrains/kotlin/kotlin-stdlib/1.3.72/kotlin-stdlib-1.3.72.jar is up-to-date (lastModified: Tue Apr 14 12:39:03 CEST 2020).

Cached resource https://jcenter.bintray.com/org/jetbrains/trove4j/trove4j/20160824/trove4j-20160824.jar is up-to-date (lastModified: Mon Feb 06 17:59:32 CET 2017).
Cached resource https://jcenter.bintray.com/org/apache/commons/commons-compress/1.12/commons-compress-1.12.jar is up-to-date (lastModified: Sat Jun 18 17:42:53 CEST 2016).
```
---

3. finally (still using **android/architecture-components-samples** project): use my PR (that removes JCenter in favor of Maven Central): https://github.com/android/architecture-components-samples/pull/959 
and (while still being in a `BasicRxJavaSample` folder/module) execute `./gradlew --refresh-dependencies` 

Build would fail :x: (becouse `jetbrains trove4j` version `20160824` can't be found anywhere... remember, JCenter is removed as a repository):

```
FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'BasicRxJavaSample'.
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not find org.jetbrains.trove4j:trove4j:20160824.
     Searched in the following locations:
       - https://dl.google.com/dl/android/maven2/org/jetbrains/trove4j/trove4j/20160824/trove4j-20160824.pom
       - https://repo.maven.apache.org/maven2/org/jetbrains/trove4j/trove4j/20160824/trove4j-20160824.pom
       - https://oss.sonatype.org/content/repositories/snapshots/org/jetbrains/trove4j/trove4j/20160824/trove4j-20160824.pom
     Required by:
         project : > com.android.tools.build:gradle:4.1.2 > com.android.tools.build:builder:4.1.2 > com.android.tools:sdk-common:27.1.2
```

File: app/build.gradle [hunk 1]
```diff
@@ -367,10 +367,7 @@ dependencies {
         exclude group: 'com.squareup.okhttp', module: 'okhttp-urlconnection'
     }
     implementation 'com.annimon:stream:1.1.8'
-    implementation ('com.takisoft.fix:colorpicker:0.9.1') {
-        exclude group: 'com.android.support', module: 'appcompat-v7'
-        exclude group: 'com.android.support', module: 'recyclerview-v7'
-    }
```
[the remark this answers is not in the document]
Comment (dejan2609) [reply app/build.gradle:377]: Exclusion mentioned above is confirmed by a `takisoft colorpicker` maintaner here: https://github.com/Gericop/ColorPicker/issues/6#issuecomment-775220802
Thank you @Gericop !

[the remark this answers is not in the document]
Comment (Gericop) [reply app/build.gradle:377]: I still need to submit the lib to Maven Central though. At least the repo is ready there, I just need to add the publish config to the lib and do the actual publishing :D

[the remark this answers is not in the document]
Comment (dejan2609) [reply app/build.gradle:377]: @Gericop Sure man, take your time ! And thank you once more for responding quickly :rocket:

**Comment by vishal206** [comment]
can I work on this issue!

**Pull request closed without merging**